config/FindXenomai.cmake

```diff
@@ -80,10 +80,38 @@ if( XENOMAI_CFLAGS_ERROR)
     message(FATAL_ERROR "Could not determine cflags with command ${XENOMAI_XENO_CONFIG} --skin=${XENOMAI_SKIN_NAME} --cflags ${XENO_CONFIG_LDFLAGS_EXTRA_ARGS}")
 endif()
 
-string(REGEX MATCHALL "-L([^ ]+)|-l([^ ]+)|.*bootstrap(-pic)?.o" XENOMAI_LIBRARY ${XENOMAI_LDFLAGS})
-string(REGEX MATCHALL "-I([^ ]+)" XENOMAI_INCLUDE_DIR ${XENOMAI_CFLAGS})
-string(REGEX MATCHALL "-D([^ ]+)" XENOMAI_COMPILE_DEFINITIONS ${XENOMAI_CFLAGS})
-string(REPLACE "-I" ";" XENOMAI_INCLUDE_DIR ${XENOMAI_INCLUDE_DIR})
+string(STRIP XENOMAI_LDFLAGS ${XENOMAI_LDFLAGS})
+string(STRIP XENOMAI_CFLAGS ${XENOMAI_CFLAGS})
+string(REPLACE " " ";" _XENOMAI_LDFLAGS ${XENOMAI_LDFLAGS})
+string(REPLACE " " ";" _XENOMAI_CFLAGS ${XENOMAI_CFLAGS})
+set(XENOMAI_LIBRARY)
+set(XENOMAI_INCLUDE_DIR)
+set(XENOMAI_DEFINITIONS)
+set(XENOMAI_OTHER_LDFLAGS)
+set(XENOMAI_OTHER_CFLAGS)
+foreach(_entry ${_XENOMAI_LDFLAGS})
+  string(REGEX MATCH "^-L(.+)|^-l(.+)|^(-Wl,.+)|^(.*bootstrap(-pic)?.o)" _lib ${_entry})
```

Comment:
Hmm, shouldn't `-L` arguments be added to `XENOMAI_LIBRARY_DIRS` instead of `XENOMAI_LIBRARY`? Passing just the dir (e.g. `/usr/lib/x86_64-linux-gnu`) to `target_link_libraries()` won't have any effect or is even illegal?

The `XENOMAI_LIBRARY_DIRS` would then have to be added by the caller with [link_directories()](https://cmake.org/cmake/help/v3.0/command/link_directories.html) or used to resolve the full path to each library using [find_library()](https://cmake.org/cmake/help/v3.0/command/find_library.html) in a second pass (preferred).

Reply:
1. To me, `XENOMAI_LIBRARY` == `XENOMAI_LIBRARY_DIRS`
2. `link_directories()` is deprecated if I recall
3. It's fine to add `-L` to `target_link_libraries` (the proof is somewhere in my mind/browser history :) )

Reply:
Ah, I was wrongly assuming that only the part of the match in parenthesis is stored in the variable `_lib`, but actually that is not true. All the matched options are forwarded to `target_link_libraries()` via `XENOMAI_LIBRARY` as-is, which is fine.